RELEASE.md

```diff
@@ -1,14 +1,59 @@
-# Stacks Release Process
+# Appsody Stacks Release Process
+A release in this repo represents a new version of a particular stack - denoted by the release tag. It will contain the following assets:
+- archives for all the templates for the released stack on [GitHub releases](https://github.com/appsody/stacks/releases)
+- latest indexes for each of the repos (`incubator`, `experimental`, `stable`) on [GitHub releases](https://github.com/appsody/stacks/releases)
+- updated docker hub images with appropriate tags on [dockerhub](https://hub.docker.com/u/appsody)
```

Comment:
nit: we could change this to: ...appropriate tags on [Appsody dockerhub](https://hub.docker.com/u/appsody)

Reply:
fixed